src/diffusers/pipelines/stable_diffusion/pipeline_stable_diffusion.py

```diff
@@ -420,17 +423,23 @@ def _encode_prompt(
 
         return prompt_embeds
 
-    def run_safety_checker(self, image, device, dtype):
-        if self.safety_checker is not None:
-            safety_checker_input = self.feature_extractor(self.numpy_to_pil(image), return_tensors="pt").to(device)
+    def run_safety_checker(self, image, device, dtype, output_type="pil"):
+        if self.safety_checker is None or output_type == "latent":
+            has_nsfw_concept = False
+        else:
+            feature_extractor_input = self.image_processor.postprocess(image, output_type="pil")
```

Comment:
```suggestion
            if torch.is_tensor(image):
                feature_extractor_input = self.image_processor.postprocess(image, output_type="pil")
            else:
                feature_extractor_input = self.numpy_to_pil(image)
```

If the image input is a numpy vector (which could be the case for backwards breaking comp) we should allow for this use case